_posts/2023-08-03-turbocharging-rust-code-verification.md

```diff
@@ -0,0 +1,445 @@
+| s2n-quic/quic/s2n-quic-core/recovery::rtt_estimator::test::weighted_average_test       | 117.62    | 129.32    | 117.62    | 9.9%   |
+| s2n-quic/quic/s2n-quic-core/varint::tests::checked_ops_test                            | 2.49      | 2.76      | 2.49      | 11.0%  |
+| s2n-quic/quic/s2n-quic-core/varint::tests::two_byte_sequence_test                      | 11.21     | 12.5      | 11.21     | 11.5%  |
+| s2n-quic/quic/s2n-quic-core/varint::tests::four_byte_sequence_test                     | 11.64     | 14.06     | 11.64     | 20.7%  |
+| s2n-quic/quic/s2n-quic-core/varint::tests::one_byte_sequence_test                      | 8.76      | 10.82     | 8.76      | 23.5%  |
+| format/fmt_u8                                                                          | 9.41      | 12.06     | 9.41      | 28.2%  |
+| s2n-quic/quic/s2n-quic-core/sync::spsc::tests::alloc_test                              | 92.84     | 126.45    | 92.84     | 36.2%  |
+| s2n-quic/quic/s2n-quic-core/random::tests::gen_range_biased_test                       | 10.48     | 14.76     | 10.48     | 40.8%  |
+| s2n-quic/quic/s2n-quic-core/xdp::decoder::tests::decode_test                           | 5.17      | 7.59      | 5.17      | 46.7%  |
+| s2n-quic/quic/s2n-quic-platform/message::msg::tests::address_inverse_pair_test         | 7.09      | 10.98     | 7.09      | 54.8   |
+|                                                                                        | **total** | **total** | **total** |        |
+|                                                                                        | 6521.99   | 1784.57   | 1676.07   |        |
+
+</div>
+</details>
+
+![Symex time](/kani-verifier-blog/assets/images/field-sens-plots/symex-time.png)
+
+We see that symbolic execution time is sometimes improved, sometimes degraded. The degradation can possibly be explained by the fact that symex has to generate more constraints and handle more basic variables with the transform activated, and spend more time applying simplifications. But this extra work has a beneficial impact on the number of symbolic execution steps and number of VCCs generated for the SAT solver, and ultimately SAT solver runtime (as seen below).
+
+![Symex steps](/kani-verifier-blog/assets/images/field-sens-plots/symex-steps.png)
+
+The number of basic symbolic execution steps is sometimes slightly higher but otherwise mostly lower with union field sensitivity activated.
+
+![number of VCCs](/kani-verifier-blog/assets/images/field-sens-plots/vccs.png)
+
+The number of verification conditions generated is sometimes slightly higher but otherwise mostly lower with union field sensitivity activated.
+
+![SAT solving time](/kani-verifier-blog/assets/images/field-sens-plots/sat-time.png)
+
+Overall the SAT solving time is sometimes slightly degraded but otherwise mostly improved across the board with field-sensitivity.
```

Comment:
```suggestion
Overall, the SAT solving time is sometimes slightly degraded but otherwise mostly improved across the board with field-sensitivity.
```